### PR TITLE
Fix thaiword_to_num to allow numbers larger than millions

### PR DESCRIPTION
**Pull request by abhabongse**
Hi maintainers,

The original implementation of the function `thaiword_to_num` works as long as the numbers does not exceed units of millions. However, 
```python
>>> thaiword_to_num("สิบห้าล้านล้าน")
6000010
>>> thaiword_to_num("หนึ่งล้านล้าน")
2000000
>>> thaiword_to_num('สองแสนสี่หมื่นสามสิบล้านสี่พันล้าน')
2244030
>>> thaiword_to_num('ร้อยสิบล้านแปดแสนห้าพัน')
10805100
```

This new code is almost a complete rewrite of the original. It now should work as expected.
```python
>>> thaiword_to_num("สิบห้าล้านล้าน")
15000000000000
>>> thaiword_to_num("หนึ่งล้านล้าน")
1000000000000
>>> thaiword_to_num('สองแสนสี่หมื่นสามสิบล้านสี่พันล้าน')
240030004000000000
>>> thaiword_to_num('ร้อยสิบล้านแปดแสนห้าพัน')
110805000
```

Please review. Feedback's welcome.

### CODE REVIEW

File: pythainlp/util/wordtonum.py [hunk 1]
```diff
@@ -2,20 +2,29 @@
...
+    r"((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)ร้อย)?"
+    r"((|หนึ่ง|ยี่|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)สิบ)?"
+    r"(|หนึ่ง|เอ็ด|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)?ล้าน)*"
+    r"(((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)แสน)?"
+    r"((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)หมื่น)?"
+    r"((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)พัน)?"
+    r"((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)ร้อย)?"
+    r"((|หนึ่ง|ยี่|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)สิบ)?"
+    r"(|หนึ่ง|เอ็ด|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)?)"
+)
```
Comment (heytitle) [inline pythainlp/util/wordtonum.py]: it would be good if we have some explanation here.

Comment (abhabongse) [reply pythainlp/util/wordtonum.py]: It is the full expansion of regexp roughly equivalent to the following EBNF grammar:
```ebnf
thaiword = { six_figures, "ล้าน" }, six_figures;
six_figures = [ [ digit ], "แสน" ], [ [ digit ], "หมื่น" ], [ [ digit ], "พัน" ], [ [ digit ], "ร้อย" ], [ [ digit ], "สิบ" ],  [ digit ];
digit = "หนึ่ง" | "สอง" | "สาม" | "สี่" | "ห้า" | "หก" | "เจ็ด" | "แปด" | "เก้า";
```
Would it be okay to put this as a comment in the code? Or would you prefer to have the code refactored to reflect this grammar pattern?

Comment (abhabongse) [reply pythainlp/util/wordtonum.py]: New code was pushed to explicitly reflect this grammar.

File: pythainlp/util/wordtonum.py [hunk 2]
```diff
@@ -9,20 +9,14 @@
...
-    r"((|หนึ่ง|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)ร้อย)?"
-    r"((|หนึ่ง|ยี่|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)สิบ)?"
-    r"(|หนึ่ง|เอ็ด|สอง|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)?)"
+_ptn_digits = r"(|หนึ่ง|เอ็ด|สอง|ยี่|สาม|สี่|ห้า|หก|เจ็ด|แปด|เก้า)"
+_ptn_six_figures = (
+    fr"({_ptn_digits}แสน)?({_ptn_digits}หมื่น)?({_ptn_digits}พัน)?"
+    fr"({_ptn_digits}ร้อย)?({_ptn_digits}สิบ)?{_ptn_digits}?"
 )
+_ptn_thai_numerals = fr"({_ptn_six_figures}ล้าน)*{_ptn_six_figures}"
+_re_thai_numerals = re.compile(_ptn_thai_numerals)
```
Comment (heytitle) [inline pythainlp/util/wordtonum.py]: 👍

**Comment by heytitle** [comment]
LGTM.

**Pull request merged**